File: 4/recursivo.c
/*
recursivo.c
*/

#include <stdio.h>
#include <time.h>

long fib(int n){

    if(n<=1)
        return n;

    return fib(n-1) + fib(n-2);
}

int main(){

    clock_t inicio = clock();

    printf("Resultado: %ld\n", fib(35));

    clock_t fin = clock();

    double tiempo = (double)(fin - inicio) / CLOCKS_PER_SEC;

    printf("Tiempo de ejecucion: %f segundos\n", tiempo);

    return 0;
}


/*
Compilar:

gcc recursivo.c

Ejecutar:

./a.out
*/

File: 4/recursivo.py
"""
recursivo.py

Versión en Python del cálculo de Fibonacci
usando recursión.

Python es interpretado, por lo que normalmente
será más lento que C en este tipo de operaciones.
"""

import time


def fib(n):

    if n <= 1:
        return n

    return fib(n-1) + fib(n-2)


inicio = time.time()

print("Resultado:", fib(35))

fin = time.time()

print("Tiempo:", fin - inicio)


"""
Ejecutar en terminal:

python3 recursivo.py
"""
